### PR TITLE
tests: minor improvements

### PR DESCRIPTION
**Pull request by jcsp**
Minor changes from while I have been working on HA tests:
- Manual pytest executions came with some warnings from `log.warn()` usage
- When something fails in a generations-enabled test, it it useful to have a log from the attachment service of what attached when, and with which generation.

### CODE REVIEW

File: test_runner/fixtures/pageserver/utils.py [hunk 1]
```diff
@@ -249,15 +249,15 @@ def assert_prefix_empty(neon_env_builder: "NeonEnvBuilder", prefix: Optional[str
...
+            log.warning(
                 f"contrading ListObjectsV2 response with KeyCount={keys} and Contents={objects}, CommonPrefixes={common_prefixes}, assuming this means KeyCount=0"
             )
             keys = 0
         elif keys != 0 and len(objects) == 0:
             # this has been seen in one case with mock_s3:
             # https://neon-github-public-dev.s3.amazonaws.com/reports/pr-4938/6000769714/index.html#suites/3556ed71f2d69272a7014df6dcb02317/ca01e4f4d8d9a11f
             # looking at moto impl, it might be there's a race with common prefix (sub directory) not going away with deletes
-            log.warn(
+            log.warning(
```
Comment (koivunej) [inline test_runner/fixtures/pageserver/utils.py:260]: I've yet to see any such local warnings. Do you have an example?

Comment (jcsp) [reply test_runner/fixtures/pageserver/utils.py:260]: They come at the end of a test run, like this:
```
[gw3] [ 40%] PASSED test_runner/regress/test_pageserver_secondary.py::test_live_migration[release-pg15]
[gw1] [ 60%] PASSED test_runner/regress/test_pageserver_secondary.py::test_location_conf_churn[release-pg15-2]
[gw0] [ 80%] PASSED test_runner/regress/test_pageserver_secondary.py::test_location_conf_churn[release-pg15-1]
[gw2] [100%] PASSED test_runner/regress/test_pageserver_secondary.py::test_location_conf_churn[release-pg15-3]

=============================== warnings summary ===============================
test_runner/regress/test_pageserver_secondary.py::test_secondary_downloads[release-pg15]
  /home/neon/neon/test_runner/fixtures/pageserver/utils.py:252: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
    log.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=================== 5 passed, 1 warning in 101.48s (0:01:41) ===================
```

File: test_runner/fixtures/log_helper.py [hunk 2]
```diff
@@ -22,6 +22,11 @@
 # log format is specified in pytest.ini file
 LOGGING = {
     "version": 1,
+    "formatters": {
+        "standard": {
+            "datefmt": "%m/%d/%Y %I:%M:%SZ %p %Z",
```
Comment (koivunej) [inline test_runner/fixtures/log_helper.py:27]:
```suggestion
            "datefmt": "%Y-%m-%d %H:%M:%SZ",
```

For the date, let's use RFC3339 as closely as possible. I am unsure how to get millis, these might still be useful for CI logs.

Surely we do not use 12h clock with pageserver logs? Allows to remove AM/PM.

If we are listing the Z, then I don't see a reason for the timezone offset (%Z) either.

Comment (koivunej) [reply test_runner/fixtures/log_helper.py:27]: I also don't understand how can this do the local time -> utc change.

File: test_runner/fixtures/pageserver/utils.py (hunk 1 above)
Comment (koivunej) [reply test_runner/fixtures/pageserver/utils.py:260]: Ok, interesting, I don't see them with python3.9. Makes sense to switch. Sadly you did not see the actual warning, which has been visible in ci only one or two times.

File: test_runner/fixtures/log_helper.py (hunk 2 above)
Comment (jcsp) [reply test_runner/fixtures/log_helper.py:27]: Hmm, yes, this is clearly broken on second look.  Weirdly, I thought it had had the desired effect but it must have been something else I did, will need to try again.

Comment (jcsp) [reply test_runner/fixtures/log_helper.py]: Ended up just dropping this commit -- it was an opportunistic thing, and in a few days my local time will be the same as UTC anyway for the winter :-D

File: control_plane/src/bin/attachment_service.rs [hunk 3]
```diff
@@ -221,8 +221,25 @@ async fn handle_attach_hook(mut req: Request<Body>) -> Result<Response<Body>, Ap
...
         });
 
-    if attach_req.pageserver_id.is_some() {
+    if let Some(attaching_pageserver) = attach_req.pageserver_id.as_ref() {
         tenant_state.generation += 1;
+        tracing::info!(
+            tenant_id = %attach_req.tenant_id,
+            "attach_hook: issuing generation {} to pageserver {}",
+            attaching_pageserver,
+            tenant_state.generation
```
Comment (koivunej) [inline control_plane/src/bin/attachment_service.rs:230]: The params are wrong way around here, text has generation before pageserver.

However, I don't think this is a good way forward even if this is a development tool.

File: control_plane/src/bin/attachment_service.rs [hunk 4]
```diff
@@ -221,8 +221,25 @@ async fn handle_attach_hook(mut req: Request<Body>) -> Result<Response<Body>, Ap
...
+        tracing::info!(
+            tenant_id = %attach_req.tenant_id,
+            "attach_hook: dropping pageserver {} in generation {}",
+            ps_id,
+            tenant_state.generation
+        );
+    } else {
+        tracing::info!(
+            tenant_id = %attach_req.tenant_id,
+            "attach_hook: no-op: tenant already has no pageserver");
```
Comment (koivunej) [inline control_plane/src/bin/attachment_service.rs:242]: We could and should add new stuff as parameters if possible, like:

```suggestion
        tracing::info!(
            tenant_id = %attach_req.tenant_id,
            ps_id = %attaching_pageserver,
            generation = %tenant_state.generation,
            "issuing",
        );
    } else if let Some(ps_id) = tenant_state.pageserver {
        tracing::info!(
            tenant_id = %attach_req.tenant_id,
            %ps_id,
            generation = %tenant_state.generation,
            "dropping",
            ps_id,
            tenant_state.generation
        );
    } else {
        tracing::info!(
            tenant_id = %attach_req.tenant_id,
            "no-op: tenant already has no pageserver");
```

Why was the "attach_hook" like a span here? Should it not be a span instead? I think we should get the spans from middleware for each path, which might already be good enough span.

**Review by koivunej — APPROVED** [review]
Minor logging changes still required, but assuming you don't return the am/pm this is good for me :)

**Pull request merged**